app/src/main/java/org/sopt/androidstudy/presentation/sign/viewmodels/SignViewModel.kt

```diff
@@ -1,17 +1,70 @@
 package org.sopt.androidstudy.presentation.sign.viewmodels
 
+import android.util.Patterns
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MediatorLiveData
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
+import java.util.regex.Pattern
 
 class SignViewModel() : ViewModel() {
     private val userEmail = MutableLiveData<String>()
     private val userPassword = MutableLiveData<String>()
-    private val isEnabledLoginButton= MutableLiveData<Boolean>()
+    private val isEnabledButton = MediatorLiveData<Boolean>()
     private val isVaildEmail = MutableLiveData<Boolean>()
-    init{
-    initEnabledLoginButton()
+    private val isVaildPassword = MutableLiveData<Boolean>()
+    private val isSuccess = MutableLiveData<Boolean>()
+
+    init {
+        initEnabledLoginButton()
+    }
+
+    private fun initEnabledLoginButton() {
+        isEnabledButton.addSource(isVaildEmail) {
+            isEnabledButton.value = combineUserVaild(it, isVaildPassword.value)
+        }
+        isEnabledButton.addSource(isVaildPassword) {
+            isEnabledButton.value = combineUserVaild(isVaildEmail.value, it)
+        }
+    }
+
+
+    fun onEmailTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {
+        userEmail.value = s.toString().trim()
+        checkEmailFormat()
+    }
+
+    fun onPasswordTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {
+        userPassword.value = s.toString().trim()
+        checkPasswordFormat()
     }
-    fun initEnabledLoginButton(){
 
+    private fun checkEmailFormat() {
+        val emailPattern = Patterns.EMAIL_ADDRESS
+        isVaildEmail.value = emailPattern.matcher(userEmail.value).matches()
     }
-}
\ No newline at end of file
+
+    private fun checkPasswordFormat() {
+        val passwordPattern =
+            Pattern.compile("^(?=.*[A-Za-z])(?=.*[0-9])(?=.*[$@$!%*#?&]).{7,15}.$")
+        isVaildPassword.value = passwordPattern.matcher(userPassword.value).matches()
+    }
+
+    private fun combineUserVaild(emailValid: Boolean?, passwordValid: Boolean?): Boolean {
+        if (isVaildEmail.value == null || isVaildPassword.value == null) {
+            return false
+        }
+        return isVaildEmail.value == true && isVaildPassword.value == true
```

Comment:
```suggestion
        if (emailValid == null || passwordValid == null) return false
        return emailValid == true && passwordValid == true
```

맞습니다! 지금 코드에서는 `initEnabledLoginButton()`에서 addSource를 한 것이 의미가 없어져버립니다 🥲
함수 내부에서 `emailValid: Boolean?, passwordValid: Boolean?` 파라미터를 이용해 주세여!! 💖